controllers/capability/capability_configmap_controller.go

```diff
@@ -0,0 +1,148 @@
+		controllerName      = "capability-configmap"
+		controllerNameShort = fmt.Sprintf("%s-controller", controllerName)
+		controllerNameLong  = fmt.Sprintf("%s/%s/%s", ctx.Namespace, ctx.Name, controllerNameShort)
+	)
+
+	r := NewReconciler(
+		ctx,
+		mgr.GetClient(),
+		ctrl.Log.WithName("controllers").WithName(controllerName),
+		record.New(mgr.GetEventRecorderFor(controllerNameLong)),
+		ctx.Namespace,
+		ctx.VMProvider,
+	)
+
+	c, err := controller.New(controllerName, mgr, controller.Options{Reconciler: r})
+	if err != nil {
+		return err
+	}
+
+	cache, err := pkgmgr.NewNamespacedCacheForObject(
+		mgr,
+		&ctx.SyncPeriod,
+		controlledType,
+		WcpClusterCapabilitiesNamespace)
+	if err != nil {
+		return err
+	}
+
+	return c.Watch(source.Kind(
+		cache,
+		controlledType,
+		&handler.TypedEnqueueRequestForObject[*corev1.ConfigMap]{},
+		predicate.TypedFuncs[*corev1.ConfigMap]{
+			CreateFunc: func(e event.TypedCreateEvent[*corev1.ConfigMap]) bool {
+				return e.Object.GetName() == WcpClusterCapabilitiesConfigMapName
+			},
+			UpdateFunc: func(e event.TypedUpdateEvent[*corev1.ConfigMap]) bool {
+				return e.ObjectOld.GetName() == WcpClusterCapabilitiesConfigMapName
+			},
+			DeleteFunc: func(e event.TypedDeleteEvent[*corev1.ConfigMap]) bool {
+				return false
+			},
+			GenericFunc: func(e event.TypedGenericEvent[*corev1.ConfigMap]) bool {
+				return false
+			},
+		},
+		kubeutil.TypedResourceVersionChangedPredicate[*corev1.ConfigMap]{},
+	))
+}
+
+func NewReconciler(
+	ctx context.Context,
+	client client.Client,
+	logger logr.Logger,
+	recorder record.Recorder,
+	vmOpNamespace string,
+	vmProvider providers.VirtualMachineProviderInterface) *Reconciler {
+	return &Reconciler{
+		Context:       ctx,
+		Client:        client,
+		Logger:        logger,
+		Recorder:      recorder,
+		vmOpNamespace: vmOpNamespace,
+		VMProvider:    vmProvider,
+	}
+}
+
+type Reconciler struct {
+	client.Client
+	Context       context.Context
+	Logger        logr.Logger
+	Recorder      record.Recorder
+	vmOpNamespace string
+	VMProvider    providers.VirtualMachineProviderInterface
+}
+
+// +kubebuilder:rbac:groups="",resources=configmaps,verbs=get;list;watch
+
+func (r *Reconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
+	ctx = pkgcfg.JoinContext(ctx, r.Context)
+
+	if req.Name == WcpClusterCapabilitiesConfigMapName && req.Namespace == WcpClusterCapabilitiesNamespace {
+		return ctrl.Result{}, r.reconcileWcpClusterCapabilitiesConfig(ctx, req)
+	}
+
+	r.Logger.Error(nil, "Reconciling unexpected object", "req", req.NamespacedName)
+	return ctrl.Result{}, nil
+}
+
+func (r *Reconciler) reconcileWcpClusterCapabilitiesConfig(ctx context.Context, req ctrl.Request) error {
+	r.Logger.Info("Reconciling WCP Cluster Capabilities Config", "configMap", req.NamespacedName)
+
+	cm := &corev1.ConfigMap{}
+	if err := r.Get(ctx, req.NamespacedName, cm); err != nil {
+		if !apierrors.IsNotFound(err) {
```

Comment:
Use `client.IgnoreNotFound(err)`

Reply:
To be clear, replace:

```go
		if !apierrors.IsNotFound(err) {
			return err
		}
		return nil
```

with:

```go
		return client.IgnoreNotFound(err)
```

The GoDoc for [`IgnoreNotFound`](https://pkg.go.dev/sigs.k8s.io/controller-runtime/pkg/client#IgnoreNotFound) explains it better:

```
IgnoreNotFound returns nil on NotFound errors. All other values that are not NotFound errors or nil are returned unmodified.
```